src/slicer.h

```diff
@@ -109,12 +116,350 @@ class SlicerLayer
 
     /*!
      * Try to close up polylines into polygons while they have large gaps in them.
-     * 
+     *
      * Clears all open polylines which are used up in the process
-     * 
+     *
      * \param[in,out] open_polylines The polylines which are stiched, but couldn't be closed into a loop yet
      */
     void stitch_extensive(Polygons& open_polylines);
+
+private:
+    /*!
+     * \brief This class represents the location of an end point of a
+     * polyline in a polyline vector.
+     *
+     * The location records the index in the polyline vector and
+     * whether this is the vertex at the start of the polyline or the
+     * vertex at the end.
+     */
+    class Terminus
+    {
+    public:
+        /*! A representation of Terminus that can be used as an array index. */
```

Comment:
I can fully understand that. In these cases it's hard to come up with good
names.

Op 25 jul. 2016 15:53 schreef "scottlenser" <notifications@github.com>:

> In src/slicer.h
> <https://github.com/Ultimaker/CuraEngine/pull/370#discussion_r72066784>:
>
> >       * \param[in,out] open_polylines The polylines which are stiched, but couldn't be closed into a loop yet
> >       */
> >      void stitch_extensive(Polygons& open_polylines);
> > +
> > +private:
> > +    /*!
> > +     * \brief This class represents the location of an end point of a
> > +     * polyline in a polyline vector.
> > +     *
> > +     * The location records the index in the polyline vector and
> > +     * whether this is the vertex at the start of the polyline or the
> > +     * vertex at the end.
> > +     */
> > +    class Terminus
> > +    {
> > +    public:
> > +        /*! A representation of Terminus that can be used as an array index. */
>
> I'll see if I can make it a little bit clearer. I'm having trouble coming
> up with good descriptions for some of the concepts even though they are
> clear in my head. I think it is because I am thinking of them spatially and
> so don't have a ready made verbal representation.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ultimaker/CuraEngine/pull/370/files/ce10cdecc7ad98f36663ad592cfbfabbca37212f#r72066784>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIe9EUF63UgPwcTOsLNOKktykMMsqvJRks5qZL_ZgaJpZM4JITdf>
> .
>